### PR TITLE
assignment 4

### PR DESCRIPTION
**Pull request by AAR0N-J**
I am just checking your code and going to give suggestions.

### CODE REVIEW

File: cst438_gradebook/src/main/resources/application.properties
Comment (AAR0N-J) [inline cst438_gradebook/src/main/resources/application.properties:1]: Add the code 
`registration.url=http://localhost:8080/course
registration.service = MQ`
under `server.port=8081`

File: cst438_gradebook/src/main/java/com/cst438/services/RegistrationServiceREST.java
Comment (AAR0N-J) [inline cst438_gradebook/src/main/java/com/cst438/services/RegistrationServiceREST.java:1]: Not super sure on this one, I thought to use `restTemplate.put(registration_url+ "/course/" + course_id + "/finalgrades", grades);` , but yours might be ok.

File: cst438_gradebook/src/main/java/com/cst438/services/RegistrationServiceMQ.java [hunk 1]
```diff
@@ -57,7 +58,13 @@ public void sendFinalGrades(int course_id, FinalGradeDTO[] grades) {
```
Comment (AAR0N-J) [inline cst438_gradebook/src/main/java/com/cst438/services/RegistrationServiceMQ.java:59]: Make sure you do not have 2 of these print statements, but its not a big deal.

Comment (AAR0N-J) [reply cst438_gradebook/src/main/java/com/cst438/services/RegistrationServiceMQ.java:59]: This print statement: `System.out.println("Start sendFinalGrades "+course_id);`

File: cst438_gradebook/src/main/java/com/cst438/services/RegistrationServiceMQ.java [hunk 2]
```diff
@@ -57,7 +58,13 @@ public void sendFinalGrades(int course_id, FinalGradeDTO[] grades) {
 		 
 		System.out.println("Start sendFinalGrades "+course_id);
 
-		//TODO convert grades to JSON string and send to registration service
+		System.out.println("Start sendFinalGrades " + course_id);
+
+	    // Convert grades to a JSON string and send it to the registration service
```
Comment (AAR0N-J) [inline cst438_gradebook/src/main/java/com/cst438/services/RegistrationServiceMQ.java]: This is all correct as far as I know but you can do it on one line too by writing `rabbitTemplate.convertAndSend(registrationQueue.getName(), asJsonString(grades));` . 
Its not necessary, but I think it makes the code look nicer.

File: cst438_gradebook/src/main/java/com/cst438/services/RegistrationServiceMQ.java [hunk 3]
```diff
@@ -45,8 +42,12 @@ public RegistrationServiceMQ() {
 	public void receive(String message) {
```
Comment (AAR0N-J) [inline cst438_gradebook/src/main/java/com/cst438/services/RegistrationServiceMQ.java:49]: This method seems good.

File: cst438_gradebook/src/main/java/com/cst438/domain/EnrollmentDTO.java [hunk 4]
```diff
@@ -2,4 +2,8 @@
 
 public record  EnrollmentDTO (int id, String studentEmail, String studentName, int courseId) {
```
Comment (AAR0N-J) [inline cst438_gradebook/src/main/java/com/cst438/domain/EnrollmentDTO.java:4]: This is ok I think

File: cst438_gradebook/pom.xml [hunk 5]
```diff
@@ -78,6 +78,11 @@
 			<scope>test</scope>
 		</dependency>
 		<!-- end selenium -->
+		<dependency>
```
Comment (AAR0N-J) [inline cst438_gradebook/pom.xml]: I do not think this is supposed to be there, but I don't know for sure

**Comment by AAR0N-J** [comment]
Looks great

**Review by katesawtell — COMMENTED** [review]
slay

**Pull request merged**